app/api/repository/route.ts

```diff
@@ -0,0 +1,54 @@
+// pages/api/repository.js
+import { db } from "@/lib/db"
+
+import { isApiKeyValid } from "../auth"
+
+export async function GET(request) {
+  if (!isApiKeyValid(request)) return
```

Comment:
should we add a check if the repository Id of the API key is the same as the requested repository ID?